tests/aws/services/sqs/test_sqs_move_task.py

```diff
@@ -0,0 +1,452 @@
+
+    response = aws_client.sqs.start_message_move_task(
+        SourceArn=source_arn, DestinationArn=destination_arn
+    )
+    snapshot.match("start-message-move-task-response", response)
+
+    # check task handle format
+    task_handle = response["TaskHandle"]
+    decoded_task_id, decoded_source_arn = decode_move_task_handle(task_handle)
+    assert uuid.UUID(decoded_task_id)
+    assert decoded_source_arn == source_arn
+
+    # check that messages arrived in destination queue correctly
+    messages = sqs_collect_messages(sqs, destination_queue, expected=2, timeout=10)
+    assert {message["Body"] for message in messages} == {"message-1", "message-2"}
+
+    # check move task completion (in AWS, approximate number of messages may take a while to update)
+    def _wait_for_task_completion():
+        _response = aws_client.sqs.list_message_move_tasks(SourceArn=source_arn)
+        assert int(_response["Results"][0]["ApproximateNumberOfMessagesMoved"]) == 2
+        return _response
+
+    response = retry(_wait_for_task_completion, retries=30, sleep=1)
+    snapshot.match("list-message-move-task-response", response)
+
+    # assert messages are no longer in DLQ
+    response = aws_client.sqs.receive_message(QueueUrl=dl_queue_url, WaitTimeSeconds=1)
+    assert not response.get("Messages")
+
+
+@markers.aws.validated
+def test_move_task_with_throughput_limit(
+    sqs_create_queue,
+    sqs_create_dlq_pipe,
+    sqs_get_queue_arn,
+    aws_client,
+    snapshot,
+):
+    sqs = aws_client.sqs
+
+    # create dlq pipe: some-queue -> dlq (source) -> destination
+    queue_url, dl_queue_url = sqs_create_dlq_pipe(max_receive_count=1)
+    source_arn = sqs_get_queue_arn(dl_queue_url)
+    destination_queue = sqs_create_queue()
+    destination_arn = sqs_get_queue_arn(destination_queue)
+
+    n = 4
+
+    # send n messages and move them into the DLQ
+    for i in range(n):
+        sqs.send_message(QueueUrl=queue_url, MessageBody=f"message-{i}")
+
+    assert sqs_wait_queue_size(sqs, queue_url, expected_num_messages=n, timeout=10) == n
+
+    # receive each message two times to move them into the dlq
+    for i in range(n * 2):
+        sqs.receive_message(QueueUrl=queue_url, VisibilityTimeout=0)
+
+    # wait until the messages arrive in the DLQ
+    assert sqs_wait_queue_size(sqs, dl_queue_url, expected_num_messages=n, timeout=10) == n
+
+    # start move task
+    response = aws_client.sqs.start_message_move_task(
+        SourceArn=source_arn, DestinationArn=destination_arn, MaxNumberOfMessagesPerSecond=1
+    )
+    snapshot.match("start-message-move-task-response", response)
+    started = time.time()
+    messages = sqs_collect_messages(sqs, destination_queue, n, 60)
+    assert {message["Body"] for message in messages} == {
+        "message-0",
+        "message-1",
+        "message-2",
+        "message-3",
+    }
+
+    assert time.time() - started >= 3
```

Comment:
I don't quite understand the time calculation/assertion here. That is mostly due to me reading through too much code right now :smile: but maybe a small comment would help (or I need a screen break)

Reply:
will add a comment! max number of messages is set to 1, so assuming there is a 1 second sleep between each message, and there are four messages, that makes at least 3 seconds :-)